lib/doorkeeper/oauth/client_credentials/creator.rb

```diff
@@ -11,7 +11,9 @@ def call(client, scopes, attributes = {})
             return existing_token
```

Comment:
Note: We are still always fetching an `existing_token`, even though it is perfectly possible, that `reuse_access_token` and `revoke_previous_client_credentials_token` are both disabled. In that case we would not really need the existing token and fetching it is just slowing us down.

I'd prefer to "hide" it behind a memoizing method, but since we need arguments from the call to `call` to obtain the token, this seems unfeasible.

What would you think about fetching it like this:

```ruby
existing_token = existing_token_for(...) if needs_existing_token?

def needs_existing_token?
  Doorkeeper.configuration.reuse_access_token || Doorkeeper.configuration.revoke_previous_client_credentials_token
end
```

Reply:
Sounds reasonable to me

Reply:
done